src/yoroi-wallets/cardano/unsignedTx.ts

```diff
@@ -211,33 +211,12 @@ const Voting = {
   }): Promise<YoroiVoting['registration']> => votingRegistration,
 }
 
-export const toDisplayAddress = async (address: string) => {
-  if ((await isByronAddress(address)) || (await isBech32Address(address))) {
-    return address
-  }
-
-  if (isBaseAddress(address) || isRewardAddress(address)) {
-    return Address.fromBytes(Buffer.from(address, 'hex')).then((address) => address.toBech32())
-  }
-
-  throw new Error('Invalid address')
-}
-
-const isByronAddress = (address: string) => ByronAddress.isValid(address)
+export const toDisplayAddress = async (address: string) =>
+  isBaseAddress(address) || isRewardAddress(address) || isEnterpriseAddress(address) || isPointerAddress(address)
+    ? Address.fromBytes(Buffer.from(address, 'hex')).then((address) => address.toBech32())
+    : address
 
-const isBech32Address = (address: string) =>
-  Address.fromBech32(address)
-    .then(() => true)
-    .catch(() => false)
-
-const isBaseAddress = (address: string) => {
-  const baseAddressRegExp = /^[0-3][0-9a-f]+$/i
-
-  return baseAddressRegExp.test(address)
-}
-
-const isRewardAddress = (address: string) => {
-  const rewardAddressRegExp = /^[ef][0-9a-f]+$/i
-
-  return rewardAddressRegExp.test(address)
-}
+const isBaseAddress = (address: string) => ['0', '1', '2', '3'].includes(address.charAt(0))
+const isPointerAddress = (address: string) => ['4', '5'].includes(address.charAt(0))
```

Comment:
`0x[4]1` = `A`, `0x[4]4` = `D`, `A` = `Byron`, `D` = `Dead/Icarus`
It needs to check for byron first, that's the only way.
https://github.com/Emurgo/yoroi-frontend/blob/9a29a38cf69b7d4472a94615c9255101b318c74e/packages/yoroi-extension/app/api/ada/lib/storage/bridge/utils.js#L277